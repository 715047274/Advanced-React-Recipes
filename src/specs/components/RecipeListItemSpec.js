import React from 'react';
import renderer from 'react-test-renderer';
import RecipeListItem from '../../components/RecipeListItem';

const testRecipe = {
  id: 1,
  name: 'Test recipe',
  category: 'Dessert',
};

describe('<RecipeListItem />', () => {
  let component;
  let tree;

  test('Should not break if no recipe passed', () => {
    expect(() => renderer.create(<RecipeListItem />)).not.toThrow();
  });

  test('Should render passed recipe', () => {
    component = renderer.create(<RecipeListItem recipe={testRecipe} />);
    tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render favorited state', () => {
    component = renderer.create(
      <RecipeListItem recipe={testRecipe} favorited />,
    );
    tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
