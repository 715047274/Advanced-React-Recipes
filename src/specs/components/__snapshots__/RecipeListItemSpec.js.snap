// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeListItem /> Should render favorited state 1`] = `
<li
  className="py2 border-bottom border-bottom-dashed pointer"
  onClick={[Function]}
>
  <span
    onClick={[Function]}
    role="button"
  >
    ✅
  </span>
  <span>
    Test recipe
  </span>
  <span>
    Dessert
  </span>
</li>
`;

exports[`<RecipeListItem /> Should render passed recipe 1`] = `
<li
  className="py2 border-bottom border-bottom-dashed pointer"
  onClick={[Function]}
>
  <span
    onClick={[Function]}
    role="button"
  >
    ⬜️
  </span>
  <span>
    Test recipe
  </span>
  <span>
    Dessert
  </span>
</li>
`;
